syft/frameworks/crypten/context.py

```diff
@@ -156,9 +168,76 @@ def _send_party_info(worker, rank, msg, return_values):
         msg (CryptenInit): message containing the rank, world_size, master_addr and master_port.
         return_values (dict): dictionnary holding return values of workers.
     """
-
+    print(f"Sending info to {rank}")
     response = worker.send_msg(msg, worker)
-    return_values[rank] = response.contents
+    print(f"Got response from {rank}")
+    return_values[rank] = _unpack_values(response.contents, model)
+    print(f"Unpacked from {rank}")
+
+
+def _pack_values(values):
+    """Pack return values to be passed into a queue then sent over the wire.
+    The main goal here is to be able to return torch tensors."""
+
+    packed_values = []
+    # single value
+    if not isinstance(values, tuple):
+        packed_values.append(_pack_value(values))
+    # multiple values
+    else:
+        for value in values:
+            packed_values.append(_pack_value(value))
+    return packed_values
```

Comment:
Quick suggestion to have recursivity to more complexe responses:
```python
def _pack_values(values):
    """Pack return values to be passed into a queue then sent over the wire.
    The main goal here is to be able to return torch tensors."""

    packed_values = []
    # multiple value
    if isinstance(values, (tuple, list)):
        for value in values:
            packed_values.append(_pack_values(values))
    # single values
    else:
        packed_values.append(_pack_value(value))
    return packed_values
```

Reply:
What I was checking here is either the function is returning multiple value (which then should be a tuple) or a single value (think `return model, loss` and `return tensor`). When 2 values gets packed, 2 should be unpacked, I'm thinking of this process more like a pipe allowing for returning remote objects, but the returned values should remain the same. Correct me if I'm wrong but this will flatten the returned values, right?